### ISSUE TITLE
Suggest Omega Mode for frequently dying players

### ISSUE THREAD

**Issue by Bugsbane** [Secretchronicles/TSC#525]
Watching my 4 year old play TSC today, I realized what a radical difference to his enjoyment of the game Omega Mode makes. He's getting good, but without it, he still would die often enough that it wasn't fun for him. The level of challenge he enjoys is just about right, just getting through the levels with Omega mode. He doesn't tend to use the jump to fly part, but it probably saves him from death every 25 seconds or so. (Literally)

It got me thinking about how I was fortunate to know about it, but most people wouldn't, and how most young kids who would otherwise enjoy the game for hours / days (as my son has) would likely give up on the game within about 5 minutes (as he would have, without Omega Mode).

So why not ask players that are clearly having trouble if they would like it enabled? Now, obviously young kids aren't going to be able to read text boxes, but if a text box comes up with a Yes / No question, they'll always just push Enter / Space, so if we make "Yes" the default, then that's handled, too.

I suggest that we ask if the player has died three times within 5 minutes without reaching a new level.

**Comment by sauer2** [comment]
@Bugsbane No offense, but could we please not target this game at 3-8 year old and take the pleasure out of it for every other player? I mean content-wise as well as difficulty level.

Also, if I remember right the plan for TSC was to throw all levels away and create new one that make sense for the concept, so I'd guess they get more balanced. This, and there was the discussion for in level savepoints.

Also you might want to reconsider putting your child in front of things he/she can't understand at 4, but anyway, your call.

**Comment by Bugsbane** [comment]
This is **not** about targeting the whole game at that age range.

This is a change that any player who would be good enough to not need omega mode would *never even see*, so it wouldn't change their experience of the game at all. If anyone did see this then, regardless of their age, they're obviously having trouble, and thus being given the *option* could make the game significantly more enjoyable for them.

Personally, I would not choose to discriminate against any player who spent many hours enjoying our game regardless of their age, when as I have said, this one change, which wouldn't impact players who didn't need it at all.

**Comment by Bugsbane** [comment]
This isn't even about my kids at all, as both switch Omega Mode on and off whenever they wish by themselves already by themselves. This is about giving a better gameplay experience to people who don't even know about omega mode.

**Comment by Quintus** [comment]
Omaga mode is intended as a cheat, and further, it is intended to help level designers test their levels. I have troubles suggesting people to use a cheat as part of regular gameplay, because that would strip the cheat from its cheat nature — a cheat is only a cheat if it can’t be discovered during normal gameplay. I agree this conflicts with the level designer help intention — maybe we should introduce a special level test mode that is only available in the editor menu.

As for frequently dying players, I rather think that this issue is only an aspect of #265. By introducing a new energy system that will not make you die instantly the frustration resulting from that will go away by itself. Then it will not be required to suggest players to use cheats, and Omega mode can remain what it is — a secret hidden cheat.

Valete,
Quintus

**Comment by sauer2** [comment]
@Bugsbane: That's not a bad thought per se, but I don't think it's 
thought through enough.

What kind of message would that send? "Booo, you suck! Have a training 
wheel skip-everything mode, don't bother learning the game properly."

Isn't that challenge what makes a jump and run game fly? You sure 
wouldn't play it for the immersion or exploration possibilities or the 
great plot.

**Comment by sydneyjd** [comment]
It is a good argument, but i honestly am against this as well. I do think it would be a good idea to make OMEGA mode easier to find though, maybe add it to the settings menu so anyone can find it? 
For me, i like having the option there, but it doesn't seem right to be able to cheat through a level.

**Comment by lambda-11235** [comment]
If the levels are too hard to progress through them, people could always play the ones in the level menu. That's what I did before I (kinda) got good. One of the main purposes of the campaign is to progress through effort and skill, and having a cheat automatically turned on would ruin.

In addition, even reasonably good players might die upwards of 5-10 times on hard levels. Asking them if they want to enable OMEGA mode might be a little insulting.

I'm personally in favor of an energy bar like @Quintus said. In addition to this we might be able to add a difficulty mode, which would change how much energy the player has. If we do this I would like an "extreme" mode which only gave me 1-2 hp.

**Comment by Quintus** [comment]
> maybe add it to the settings menu so anyone can find it? 

No. It is a cheat, and not for regular gameplay. As I outlined above, I would rather add a special "level testing mode" accessible from the editor menu to make level designer’s life easier so that Omega mode is not used so frequently and remains a rather secret cheat.

> In addition to this we might be able to add a difficulty mode, which would change how much energy the player has.

I ilke that idea. The discussion about this should however rather belong to #265.

Valete,
Quintus

**Comment by datahead8888** [comment]
> I suggest that we ask if the player has died three times within 5 minutes without reaching a new level.

With this set up, more experienced players might end up seeing it in more difficult levels, when we expand the game to have a more challenging final world.

> Omaga mode is intended as a cheat, and further, it is intended to help level designers test their levels. I have troubles suggesting people to use a cheat as part of regular gameplay, because that would strip the cheat from its cheat nature

I tend to agree with this argument.  I have seem some games reveal cheats after beating the game, though this is different.

> could we please not target this game at 3-8 year old and take the pleasure out of it for every other player?

> This is a change that any player who would be good enough to not need omega mode would never even see

No, most players wouldn't see it (as long as we figured out how to handle harder levels with this feature), but if people heard we built the game this way (ie. forum discussions), it might give the impression that the game is targeted more towards children than adults.

> By introducing a new energy system that will not make you die instantly the frustration resulting from that will go away by itself.

Introducing a life energy system will certainly help moderate difficulty.  I think we also need to talk more about instant death in pits / lava (or power ups that make it easier to dodge these regions), but that is an issue for #265.

>  In addition to this we might be able to add a difficulty mode

We've talked about having a "Kids" campaign for kids (as well as an "expert" campaign for advanced players).  Full, new campaigns with different levels have the advantage of offering more experiences if we can build them.  A difficulty setting might make for a good interim solution, however, since it takes a lot less time to develop.  The player could take less damage when hit by enemies, once a life bar is added.  We could then talk about switching to more campaigns later.

> You sure wouldn't play it for the immersion or exploration possibilities or the great plot.

There is no reason we cannot have a good plot.  It is more of a question of how to break tradition and present that plot.  I am sorry the story development has been going slowly, however.
